### ISSUE TITLE
Component structure is confusing

### ISSUE THREAD

**Issue by AlbertoALopez** [Pairboard/Pairboard#9]
I've seen it mentioned a few times that the current component structure can be a bit confusing. Would it be helpful to change the structure slightly to make it more intuitive for those contributing? 

One approach would be to group components by what context they're used in. For example you could have:

src/
---- components/
--------- Header/
------------- Header.jsx
------------- Header.test.js
--------- CamperList/
------------- CamperList.jsx
------------- CamperCard.jsx
------------- CamperList.test.js
------------- CamperCard.test.js
--------- Login/
------------- LoginModal.jsx
------------- LoginModal.test.js
--------- Footer/
------------- Footer.jsx
------------- Footer.test.js
etc.

You could also group by what would map closer to views in a traditonal MVC application.

src/
---- views/
--------- Home/ <- main route or landing page
------------- Header/ <- these are always there but make sense to
------------------- Header.jsx
------------------- Header.test.js
------------- Footer/
------------------- Footer.jsx
------------------- Footer.test.js
--------- Campers/
------------- CamperList/
------------- CamperCard/
--------- Login/
etc.

Basically you can get however complicated you want to when it comes to file structure. There are lots of articles online each with their own opinion about what's best. I like this one [http://jaysoo.ca/2016/02/28/organizing-redux-application/](http://jaysoo.ca/2016/02/28/organizing-redux-application/) which deals with Redux as well as React and this one: [http://marmelab.com/blog/2015/12/17/react-directory-structure.html](http://marmelab.com/blog/2015/12/17/react-directory-structure.html).

It would be nice to at least have related files grouped together with their tests and be able to tell at a glance if it is a container or presentational component.

Let me know what you guys think and what would be most intuitive to you!
